cmake/LLVMIRUtil.cmake

```diff
@@ -175,16 +175,17 @@ function(llvmir_attach_bc_target)
       continue()
     endif()
 
+    get_target_property(SOURCE_DIR ${DEPENDS_TRGT} SOURCE_DIR)
+    get_filename_component(INFILE ${IN_FILE} ABSOLUTE BASE_DIR ${SOURCE_DIR})
     get_filename_component(OUTFILE ${IN_FILE} NAME)
-    get_filename_component(INFILE ${IN_FILE} ABSOLUTE)
     set(OUT_LLVMIR_FILE "${OUTFILE}.${LLVMIR_BINARY_FMT_SUFFIX}")
     set(FULL_OUT_LLVMIR_FILE "${WORK_DIR}/${OUT_LLVMIR_FILE}")
 
     # compile definitions per source file
-    llvmir_extract_compile_defs_properties(IN_FILE_DEFS ${IN_FILE})
+    llvmir_extract_compile_defs_properties(IN_FILE_DEFS ${INFILE})
```

Comment:
Ideally, I'd like to use only one version of the "infile" as derived from any path processing of `IN_FILE` which changes at the `foreach`.
Can this be done?
Maybe name it `ABS_IN_FILE` or something similar?